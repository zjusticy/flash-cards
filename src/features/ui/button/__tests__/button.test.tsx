import React from 'react';
// import renderer from "react-test-renderer";
// import configureStore from 'redux-mock-store'
// import { shallow } from 'enzyme';
// import toJson from 'enzyme-to-json';
import { afterEach, describe, it, expect } from 'vitest';

import { render, cleanup } from '@testing-library/react';

import Button from '../button';

afterEach(cleanup);

describe('Button', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <Button btnType="Success" size="Big" elementType="submit">
        Facebook
      </Button>
    );
    // expect(getByRole("button").classList.contains("Big")).toBe(true);
    // expect(getByRole("button").textContent).toEqual("Facebook");

    expect(asFragment()).toMatchSnapshot();
  });

  // it("Button changes the class when type changed 2", () => {
  //   const { getByRole } = render(
  //     <Button btnType="Success" size="Medium" elementType="submit">
  //       Facebook
  //     </Button>
  //   );
  //   expect(getByRole("button").classList.contains("Medium")).toBe(true);
  // });

  // it("Button changes the class when type changed 3", () => {
  //   const { getByRole } = render(
  //     <Button btnType="Success" size="Small" elementType="submit">
  //       Facebook
  //     </Button>
  //   );
  //   expect(getByRole("button").classList.contains("Small")).toBe(true);
  // });

  // it("Button changes the class when type changed", () => {
  // 	render(
  // 		<Button btnType="Success" size="Medium" elementType="submit">
  // 			Facebook
  // 		</Button>
  // 	);
  // 	expect(getByRole('button').classList.contains('Medium')).toBe(true);
  // });

  // // manually trigger the callback
  // tree.props.onMouseLeave();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});
