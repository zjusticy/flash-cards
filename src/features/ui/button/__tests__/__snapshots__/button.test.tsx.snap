// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders correctly 1`] = `
<DocumentFragment>
  <button
    class="bg-transparent cursor-pointer font-bold font-pressStart duration-100 disabled:color-[#ccc] disabled:cursor-not-allowed [&_a]:inline-block [&_a]:no-underline [&_a]:h-full [&_a]:w-full [&_a]:align-middle [&_a]:text-center [&_a]:font-pressStart outline-none border-2 border-solid border-button-color hover:bg-button-color hover:text-secondary-color [&_a]:hover:text-secondary-color [&_a]:text-button-color text-button-color text-[1rem] h-16 px-4 py-2 [&_a]:text-[1rem]"
    type="submit"
  >
    Facebook
  </button>
</DocumentFragment>
`;

exports[`Button renders correctly 1`] = `
<DocumentFragment>
  <button
    class="Button Success Big "
    type="submit"
  >
    Facebook
  </button>
</DocumentFragment>
`;
